server/ladder_service.py

```diff
@@ -249,89 +268,99 @@ async def inform_player(self, player: Player):
+                ])
+                asyncio.create_task(
+                    self.start_game(s1.players, s2.players, queue.name)
                 )
-                asyncio.create_task(self.start_game(p1, p2))
             except Exception as e:
                 self._logger.exception(
                     "Error processing match between searches %s, and %s: %s",
                     s1, s2, e
                 )
 
-    async def start_game(self, host: Player, guest: Player):
+    async def start_game(self, team1: List[Player], team2: List[Player], queue_name: str):
         # TODO: Get game_mode from queue
+        self._logger.debug(
+            "Starting %s game between %s and %s", queue_name, team1, team2
+        )
+        game_mode = "faf" if queue_name == "ladder2v2" else queue_name
+
         try:
-            self._logger.debug(
-                "Starting ladder game between %s and %s", host, guest
-            )
+            host = team1[0]
+            all_players = team1 + team2
+            all_guests = all_players[1:]
+
             host.state = PlayerState.HOSTING
-            guest.state = PlayerState.JOINING
+            for guest in all_guests:
+                guest.state = PlayerState.JOINING
 
             played_map_ids = await self.get_game_history(
-                [host, guest],
+                all_players,
                 "ladder1v1",
                 limit=config.LADDER_ANTI_REPETITION_LIMIT
             )
             rating = min(
                 newbie_adjusted_ladder_mean(player)
-                for player in (host, guest)
+                for player in all_players
             )
             pool = self.queues["ladder1v1"].get_map_pool_for_rating(rating)
             if not pool:
                 raise RuntimeError(f"No map pool available for rating {rating}!")
             map_id, map_name, map_path = pool.choose_map(played_map_ids)
 
+            # TODO: Different game mode for team matchmaker?
             game = self.game_service.create_game(
-                game_mode='ladder1v1',
+                game_mode=game_mode,
                 host=host,
-                name=f"{host.login} Vs {guest.login}"
+                name=game_name(team1, team2)
             )
-
-            host.game = game
-            guest.game = game
-
+            game.init_mode = InitMode.AUTO_LOBBY
             game.map_file_path = map_path
 
-            game.set_player_option(host.id, 'StartSpot', 1)
-            game.set_player_option(guest.id, 'StartSpot', 2)
-            game.set_player_option(host.id, 'Army', 1)
-            game.set_player_option(guest.id, 'Army', 2)
-            game.set_player_option(host.id, 'Faction', host.faction.value)
-            game.set_player_option(guest.id, 'Faction', guest.faction.value)
-            game.set_player_option(host.id, 'Color', 1)
-            game.set_player_option(guest.id, 'Color', 2)
+            for i, player in enumerate(alternate(team1, team2)):
```

Comment:
Right, I forgot to add the check to skip `None` values. In practice I don't expect that it will ever happen that way. The matchmaker queue should probably always spit out even teams, but it's barely any extra code to make it more general, so I'm doing it :P.